### PR TITLE
Matrix decompose

### PR DESCRIPTION
**Pull request by CodeDoes**
Can rotation be done better?
This

### CODE REVIEW

File: glm/mat_transform.nim [hunk 1]
```diff
@@ -91,6 +92,21 @@ proc scale*[T](m:Mat4[T], s: T): Mat4[T] {.inline.} =
...
+## DECOMPOSE
+proc pos*[T](m: Mat4[T]): var Vec3[T] =
+  m.row(0).xyz
+proc `pos=`*[T](m: var Mat4[T], val: Vec3[T]) =
+  m.row0 = vec4f(val,1)
+proc scale*[T](m: Mat4[T]): var Vec3[T] =
+  m.diag.xyz
+proc scale*[T](m: var Mat4[T], val: Vec3[T]) =
+  m.diag = vec4f(val,1)
+proc rot*[T](m: Mat4[T]): var Quat =
```
Comment (krux02) [inline glm/mat_transform.nim]: This does not return a ``var Quat``.

**Comment by krux02** [comment]
can you point me to the c++ glm version of the other functions you added? The diag proc I like a lot, but the others, I am not so sure about them. You know I am the maintainer of this repository and as soon as I accept this PR they become my liability, and I would really like to know in advance that they won't cause problems. So can you create an issue for why you would want these procedures in glm? How do you want to use them?

**Comment by CodeDoes** [comment]
I want to get the position. at the moment it needs me to know exactely where they are. If there other/better ways to get the same properties then I'd rather use that. 
There's this, which is experimental but I still need it or something similar. 
https://glm.g-truc.net/0.9.6/api/a00204.html

***How do you want to use them?***
I want to be able to set the position,scale,rotation,skew independently without having to reference the source to see which rows and columns contain what.

**Comment by krux02** [comment]
well, when you want to access the position scale and rotation really independently, I recommend you to use a data type like this  ``tuple[position: Vec4f; scale: Vec4f; rotation: Quatf]``. A matrix doesn't store thise values independently. The scale isn't the diagonal part of a matrix, that is only true for a pure scale matrix, but as soon as you also add a rotation to the matrix it breaks. Also the position has no fixed location. Depending on if you prefer to multiply your matrix from right or from left, the position is either in the last row or in the last column. DirectX prefers the last row and OpenGL prefers the last column.  Because this library is basically a clone of the OpenGL shading language features I would highly recommend the last column. But also the translation is not independently stored of the rotation.

In other words. I cant accept the decompose like this. A direct copy paste port of c++ glm decompose would be ok thoguh.

**Comment by krux02** [comment]
Since this is not something I would pull. I am closing this PR now. If you decide to port the glm gtx function, you can still open a new pull request.
